wandb/sdk/launch/utils.py

```diff
@@ -434,8 +435,30 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
                 "Error: %s" % (version, uri, e)
             )
     else:
-        repo.create_head("master", origin.refs.master)
-        repo.heads.master.checkout()
+        # Check if main is in origin, else set branch to master
+        branches = [ref.name for ref in repo.references]
+        if "main" in branches or "origin/main" in branches:
+            version = "main"
+        else:
+            version = "master"
+
+        try:
+            repo.create_head(version, origin.refs[version])
+            repo.heads[version].checkout()
+            wandb.termlog(f"No git branch passed. Defaulted to branch: {version}")
+        except git.exc.GitCommandError as e:
+            raise ExecutionError(
```

Comment:
Do we want `LaunchErrors` here instead of `ExecutionErrors`? @KyleGoyette

Reply:
Yep, throw a launch error here, the agent catches it better.